pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -138,10 +138,17 @@ func (pl *DefaultPreemption) preempt(ctx context.Context, state *framework.Cycle
 	}
 
 	// 2) Find all preemption candidates.
-	candidates, err := pl.FindCandidates(ctx, state, pod, m)
-	if err != nil || len(candidates) == 0 {
+	candidates, nodeNum, status, err := pl.FindCandidates(ctx, state, pod, m)
+	if err != nil {
 		return "", err
 	}
+	if len(candidates) == 0 {
+		return "", &framework.FitError{
```

Comment:
It's not clear to me that how this returned obj is utilized by subsequent PostFilter plugins.

Reply:
This PR intends to change the public method of `FindCandidates` to return node `status`, so that the calling of this method could get the node status as well.

Quote from @ahg-g 's initial comments, 
> This is just so external preemption plugins that use this interface can benefit from this info, but this is less critical for this PR, so I guess we can leave this here for now.


As to the return `FitError` here, it's used to populate the log message on the preemption failure.  I did some refactoring to move this obj from  line 224 to here, the reason is I need to add the `unschedulableNodeStatus` into the status or else user might confuse that they have for example they have 3 nodes in the cluster, but only give hint on one node that doesn't have enough resource but no idea on what's happening on other two nodes (please see my update on the testcases).

In short, it is sill used by log population.

Reply:
> so that the calling of this method could get the node status as well.

Well, the current version is not stateless so arguably I doubt if users would like to use it... But I see your point.